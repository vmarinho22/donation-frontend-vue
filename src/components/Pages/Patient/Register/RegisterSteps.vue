<template>
  <v-container fluid>
    <linear-stepper :steps="steps" :on-finish="onFinishSignUp" />
  </v-container>
</template>

<script setup lang="ts">
import LinearStepper from '@/components/Steppers/LinearStepper/LinearStepper.vue'
import StepOne from './StepOne.vue';
import StepTwo from './StepTwo.vue';
import { useI18n } from 'vue-i18n';

const { t } = useI18n()

const steps = [
  {
    title: t('pages.patient.register.stepOne'),
    component: StepOne
  },
  {
    title: t('pages.patient.register.stepTwo'),
    component: StepTwo
  }
]

async function onFinishSignUp() {
  console.log('finish')
}

</script>