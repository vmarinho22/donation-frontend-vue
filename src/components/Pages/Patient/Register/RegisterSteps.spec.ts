import { describe, it, expect, vi } from 'vitest'

import { shallowMount } from '@vue/test-utils'
import RegisterSteps from './RegisterSteps.vue'
import vuetify from '@/tests/plugins/vuetify'
import { createTestingPinia } from '@pinia/testing'

vi.mock('vue-i18n', () => ({
  useI18n: vi.fn().mockReturnValue({
    t: vi.fn().mockReturnValue('translation')
  })
}))

describe('RegisterSteps', () => {
  it('renders the component', () => {
    const wrapper = shallowMount(RegisterSteps, {
      global: {
        plugins: [vuetify, createTestingPinia({
          createSpy: vi.fn()
        })]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
