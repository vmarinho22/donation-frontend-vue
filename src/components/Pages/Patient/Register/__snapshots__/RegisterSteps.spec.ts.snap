// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RegisterSteps > renders the component 1`] = `"<linear-stepper-stub steps=\\"[object Object],[object Object]\\"></linear-stepper-stub>"`;
